### PR TITLE
OSDOCS-5199:AWS EFS cross account support

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.13+

Issue: https://issues.redhat.com/browse/STOR-990

Link to docs preview: https://56409--docspreview.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-aws-efs.html#persistent-storage-csi-efs-cross-account_persistent-storage-csi-aws-efs

QE review:
- [x] QE has approved this change.

Additional information:

**PTAL**: @jsafrane, @ropatil010, @gcharot

### CODE REVIEW

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        "Effect": "Allow",
+        "Action": "sts:AssumeRole",
+        "Resource": "arn:aws:iam::589722580343:role/test-efs-acrossaccount-role"
+    }
+}
+----
+
+. In AWS account A, attach the AWS-managed policy “AmazonElasticFileSystemClientFullAccess” to {product-title} cluster master role. The role name is in the form `<clusterID>-master-role` (for example, `test-0120ef-czjrl-master-role`).
+
+. Create a Kubernetes secret with "awsRoleArn" as the key and the role created earlier as the value:
```
Comment (GroceryBoyJr) [inline modules/persistent-storage-csi-efs-cross-account.adoc:212]: Suggestion: replace doublequotes with backtick: `awsRoleArn` to indicate a value to be included in the oc command.

**Review by GroceryBoyJr — COMMENTED** [review]
@lpettyjo only a few small nits. Great work!

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 2]
```diff
@@ -0,0 +1,304 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-csi-aws-efs.adoc
+// 
+
+:_content-type: PROCEDURE
+[id="persistent-storage-csi-efs-cross-account_{context}"]
+= AWS EFS CSI cross account support
```
Comment (gcharot) [inline modules/persistent-storage-csi-efs-cross-account.adoc:8]: Looking at the preview, this chapter is placed after "Uninstalling the AWS EFS CSI Driver Operator" shouldn't it be placed before?

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+Cross account support allows you to have an {product-title} cluster in one AWS account and mount your file system in another AWS account using the AWS Elastic File System (EFS) Container Storage Interface (CSI) driver.
+
+[NOTE]
+====
+Both the {product-title} cluster and EFS file system should be in the same region.
```
Comment (gcharot) [inline modules/persistent-storage-csi-efs-cross-account.adoc:14]: must be in the same region?

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+* Mount an AWS EFS file system in account B
+
+To use AWS EFS across accounts:
+
+. Install {product-title} cluster with AWS account A and install the EFS CSI Driver Operator.
+
+. Create an EFS volume in AWS account B:
+
+.. Create a test virtual private cloud (VPC) called “test-efs-vpc” with CIDR “172.20.0.0/16” and subnet for the AWS EFS volume.
```
Comment (gcharot) [inline modules/persistent-storage-csi-efs-cross-account.adoc:37]: i would avoid using the term "test" in our naming convention.
Also it's up to the customer to choose his CIDR, 172.20.0.0/16 is just and example. The wording suggests it has to be set to this value. thoughts?

File: modules/persistent-storage-csi-efs-cross-account.adoc (hunk 2 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-efs-cross-account.adoc:8]: Moved it earlier.

File: modules/persistent-storage-csi-efs-cross-account.adoc (hunk 4 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-efs-cross-account.adoc:37]: I'm pretty sure you don't need that value. I've indicated that it is an example.

File: modules/persistent-storage-csi-efs-cross-account.adoc (hunk 3 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-efs-cross-account.adoc:14]: @gcharot According QE's test procedure: https://docs.google.com/document/d/1-c87alz1p4Q6xUxIEx-aogkocZpaXeLXbN3HVrxLvPo/edit#

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+. Create an EFS volume in AWS account B:
+
+.. Create a virtual private cloud (VPC) called, for example, "my-efs-vpc” with CIDR, for example, “172.20.0.0/16” and subnet for the AWS EFS volume.
+
+.. On the AWS console, go to https://console.aws.amazon.com/efs.
+
+.. Click *Create new filesystem*:
+
+... Create a file named, for example, "my-filesystem”.
```
Comment (jsafrane) [inline modules/persistent-storage-csi-efs-cross-account.adoc:43]: Create a *filelesystem* named

File: modules/persistent-storage-csi-efs-cross-account.adoc (hunk 3 above)
Comment (gcharot) [reply modules/persistent-storage-csi-efs-cross-account.adoc:14]: yes i think this is just an abuse of the word "should". We don't support multi region so I assume "must" is better suited?

Comment (lpettyjo) [reply modules/persistent-storage-csi-efs-cross-account.adoc:14]: Correct. Done.

**Comment by jsafrane** [comment]
lgtm

File: modules/persistent-storage-csi-efs-cross-account.adoc [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+            "Principal": {
+                "AWS": "arn:aws:iam::301721915996:root"
+            },
+            "Action": "sts:AssumeRole",
+            "Condition": {}
+        }
+    ]
+}
+
+# Trust relationships trusted entity trusted account A configuration on my-efs-acrossaccount-role in account B
```
Comment (ropatil010) [inline modules/persistent-storage-csi-efs-cross-account.adoc:131]: @lpettyjo Can you plz remove Line L131 to L145. Bec L115 to L129 are same with L131 to L145/added two times.
Thanks. Others LGTM.

Comment (lpettyjo) [reply modules/persistent-storage-csi-efs-cross-account.adoc:131]: Ha. Good catch indeed. Fixed.

**Pull request merged**